core_impl/src/test/java/io/opencensus/implcore/stats/ViewManagerImplTest.java

```diff
@@ -429,31 +477,38 @@ public void testRecordWithEmptyStatsContext() {
   }
 
   @Test
-  public void testRecordWithNonExistentMeasurement() {
+  public void testRecord_MeasureNameNotMatch() {
+    testRecord_MeasureNotMatch(
+        MeasureDouble.create(MEASURE_NAME, "measure", MEASURE_UNIT),
+        MeasureDouble.create(MEASURE_NAME_2, "measure", MEASURE_UNIT),
+        10.0);
+  }
+
+  @Test
+  public void testRecord_MeasureTypeNotMatch() {
+    testRecord_MeasureNotMatch(MEASURE_LONG, MEASURE_DOUBLE, 10.0);
```

Comment:
Suggestion: This test would be more thorough if it used the same name for both measures.  testRecord_MeasureNameNotMatch already checks that measures with different names don't interfere.

Reply:
Correct, I also updated MeasureToViewMap for checking duplicate measure names.